ethsigner/src/main/java/tech/pegasys/ethsigner/requesthandler/sendtransaction/NoncePopulatingRawTransactionSupplier.java

```diff
@@ -0,0 +1,45 @@
+/*
+ * Copyright 2019 ConsenSys AG.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not use this file except in compliance with
+ * the License. You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software distributed under the License is distributed on
+ * an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the License for the
+ * specific language governing permissions and limitations under the License.
+ */
+package tech.pegasys.ethsigner.requesthandler.sendtransaction;
+
+import java.io.IOException;
+
+import org.web3j.crypto.RawTransaction;
+
+public class NoncePopulatingRawTransactionSupplier implements RawTransactionSupplier {
+
+  private final NonceProvider nonceProvider;
+  private final RawTransaction template;
+
+  public NoncePopulatingRawTransactionSupplier(
+      final NonceProvider nonceProvider, final RawTransaction template) {
+    this.nonceProvider = nonceProvider;
+    this.template = template;
+  }
+
+  @Override
+  public RawTransaction get() throws IOException {
+    return RawTransaction.createTransaction(
+        nonceProvider.getNonce(),
+        template.getGasPrice(),
+        template.getGasLimit(),
+        template.getTo(),
+        template.getValue(),
+        template.getData());
+  }
+
+  @Override
+  public String toString() {
+    return "NoncePopulatingRawTransactionSupplier{" + "template=" + template + '}';
```

Comment:
You can merge two of the string: `"NoncePopulatingRawTransactionSupplier{template=" + template + '}';`

Reply:
File removed.